facades/PC/build.gradle.kts

```diff
@@ -217,103 +239,78 @@ tasks.register<Copy>("createVersionFile") {
     filter(FixCrLfFilter::class, "eol" to FixCrLfFilter.CrLf.newInstance("crlf"))
 }
 
-// TODO: This could probably be done more Gradley (engine project resource dir instead of direct path?) and with some variables
-tasks.register<Copy>("copyCreditsFile") {
-    description = "Copies the credits file into the engine's resource dir where it'll be read at runtime"
-    from("$rootDir/docs")
-    into("$rootDir/engine/src/main/resources")
-    include("Credits.md")
-}
-
-// Main application dist target. Does NOT include any modules.
-tasks.register<Sync>("distApp") {
-    description = "Creates an application package for distribution"
+tasks.register<Zip>("distForLauncher") {
     group = "terasology dist"
+    description = "Bundles the project to a Launcher-compatible layout."
 
-    dependsOn("createVersionFile")
-    dependsOn("copyCreditsFile")
-    dependsOn(":extractNatives")
-    dependsOn("jar")
-
-    into("${distsDirectory.get().asFile}/app")
-    from ("$rootDir/README.markdown") {
-        filter(FixCrLfFilter::class, "eol" to FixCrLfFilter.CrLf.newInstance("crlf"))
-        rename("README.markdown", "README")
-    }
-    from ("$rootDir/LICENSE") {
-        filter(FixCrLfFilter::class, "eol" to FixCrLfFilter.CrLf.newInstance("crlf"))
-    }
-    from ("$rootDir/NOTICE") {
-        filter(FixCrLfFilter::class, "eol" to FixCrLfFilter.CrLf.newInstance("crlf"))
-    }
-    from("launchScripts") {
-        exclude("TeraEd.exe")
-    }
-
-    from("$buildDir/$versionFileName") {}
+    archiveFileName.set("Terasology.zip")
 
-    into(subDirLibs) {
-        from(configurations.runtimeClasspath)
-        from(tasks.getByPath(":engine:jar"))
-        from("$buildDir/libs") {
-            include("*.jar")
-            rename {
-                "Terasology.jar"
+    // Launcher expects `libs/Terasology.jar`, no containing folder
+    // TODO: fix launcher so it can take either structure. It should be able to do without ambiguity.
+    val defaultLibraryDirectory = "lib"
+    val launcherLibraryDirectory = "libs"
+
+    this.with(distributions.getByName("main").contents {
+        eachFile {
+            val pathSegments = relativePath.segments
+
+            when (pathSegments[0]) {
+                defaultLibraryDirectory -> {
+                    // Redirect things from lib/ to libs/
+                    val tail = pathSegments.sliceArray(1 until pathSegments.size)
```

Comment:
This looks pretty fancy. I trust the comment but have no idea how the magic works.

Reply:
I was disappointed with how verbose my Kotlin solution for this was. This

https://github.com/MovingBlocks/Terasology/blob/9d84f8ee66ba268d0a49ce48dc8f4287027cf3d8/facades/PC/build.gradle.kts#L259-L261

would be something like this in Python:

```python
relativePath = RelativePath(true, launcherLibraryDirectory, *pathSegments[1:])
```

but I didn't figure out a way to do that kind of open-ended slice in Kotlin. I tried just `.drop()`ing the first segment, but that left me with a List instead of an Array and I seem to need Array to use the spread operator with the varargs argument.